### DISCUSSION TITLE
Constraints for input flows from reserve nodes

### DISCUSSION THREAD

**Discussion by datejada** [spine-tools/SpineOpt.jl#796] [Ideas]
Continue in this discussion the topic of constraints that new changes to consider correctly the consumer units providing reserve. See, https://github.com/spine-tools/SpineOpt.jl/discussions/788#discussioncomment-7553629

**Comment by manuelma** [comment]
Nice one @datejada - so if I may ask, ideally we would list the necessary changes with respect to this overleaf document https://www.overleaf.com/project/6454fde7fdf7f1c73ceeb2a0

**Reply by g-moralesespana** [reply]
> @manuelma  https://www.overleaf.com/project/6454fde7fdf7f1c73ceeb2a0 - maybe highlight the new terms in red or something. I can work on that in a new PR.

The overleaf document applies exactly how it is for unit output flows. For unit input, all upwards reserves change to downwads; and all downwards change to upwards. That is, following the overleaf nomenclature, all the "+" superscripts change to "-"; and all the "-" superscripts change to "+".

**Reply by manuelma** [reply]
So that's the only thing? Sounds perfect, I will get to it then, thanks @g-moralesespana

**Reply by manuelma** [reply]
Ok so the above is clear @g-moralesespana - but what about the u^{N-} variables, do they need to be switched to u^{N+} for constraints involving input flows? My guess is no, except for maybe equation 7, lower bound on the unit flow. Am I right?

Sorry for so many questions, but I'm simply unable to make my own mind about this stuff.

**Comment by manuelma** [comment]
I also changed the title so it's clearer from SpineOpt's perspective,  "consumption units providing reserves" => "input flows from reserve nodes".

**Comment by nhniina** [comment]
Just something to keep in mind: if we keep the minimum reserve activation time constraint(s), they should also be made to work with input flows from reserve nodes (output flows to storage nodes).

**Reply by g-moralesespana** [reply]
I added the reserves to the storage constraints: https://www.overleaf.com/project/6454fde7fdf7f1c73ceeb2a0

**Comment by manuelma** [comment]
@g-moralesespana @datejada @gnawin can we review one thing again?

Take equation 4 in the overleaf https://www.overleaf.com/project/6454fde7fdf7f1c73ceeb2a0 for the unit flow upper bound. Yesterday we said that to convert that equation to 'input flows', we need to switch the plus to minus in the reserve indices. 

But I think we also need, on the RHS, to move the u^{N+}_t to the third term so it gets multiplied by (F - SU) rather than (F - SD). This is because the flow for non-spinning units *started up* should be constrained by SU rather than SD. Does this make any sense or I'm just missing the entire picture here?

**Reply by datejada** [reply]
We can keep the RHS of the equations the same as the initial changes we discussed. The reason is that from the system point of view, equation (4), when it is a consumption unit, providing $u^{N+}_t$ means that the unit is shutting down (i.e., the other way around of production units). Therefore, multiplying by $(F - SD)$ is correct.

The confusion arises due to naming initially created for production units. But now, with the consumption units feature, the names in the formulation create a little confusion, and you need to remember that the concept is flipped.

**Reply by manuelma** [reply]
Ahhh, I think I get it. Thank you very much! (indeed, the naming was confusing me).
